pypesto/ensemble/ensemble.py

```diff
@@ -172,8 +180,13 @@ def _stack_outputs(ic: int):
             if self.prediction_results[0].conditions[ic].output is None:
                 return None
             # stack predictions
-            output_list = [prediction.conditions[ic].output
-                           for prediction in self.prediction_results]
+            if isinstance(self.prediction_results[0].conditions[ic].output,
+                          np.ndarray):
+                output_list = [prediction.conditions[ic].output
+                               for prediction in self.prediction_results]
+            else:
+                output_list = [prediction.conditions[ic].output[AMICI_Y]
```

Comment:
I just saw below, that it seems you intend to change the type of `prediction.conditions[ic].output`. However, this is a data member of the class PredictionConditionResult. I'd strongly argue that data members of classes should not switch type, if possible, and if inevitable: Only during major refactorings, as this will typically break backwards compatibility. Rather add a new data member for the weights...